### ISSUE TITLE
How to manage transactions that are tranfsers?

### ISSUE THREAD

**Issue by NanneHuiges** [firefly-iii/firefly-iii#1845]
**TL;DR**: When I wire money from one account to another it should be a transfer in the end. Currently it comes up as 2 transactions. How can I deal with this best?

_I am running Firefly III version 4.7.7_

When importing (both from ABN and from ING), transfers are imported as transactions. The numbers match up in the end, but the 'balance' , income vs expenses, is inflated because of this. If I transer some number x from one account to the other, the one account gets extra expenses, and the other extra income. So in the end the final numbers are correct, but I believe it is meant to be a transfer.

I have found out from the issues here that I can change a transaction to a transfer, and have done so, but it seems that I have to change one transaction to a transfer, but then need to remove the second transaction (I import from both banks, so if I move x euro from abn to ing then I can make the negative transaction from abn into a transfer, but need to delete the positive one from the ing). This is tricky because if I do a broad import, it will get re-added sadly

Is this the way to do transfers between banks, or deposits to a savings account?

**Steps to reproduce** 
* Wire money from one of your asset accounts to another
* import transactions from the first account (you get an expense here)
* import transactions from the second account (you get income here)
* notice the inflated numbers for your balance (income/expense), and the correct final account-balance
* change one of the transactions into a transfer
* notice the wrong balances
* remove the other transaction
* notice how all seems good
* (for bonus points: import the transactions where all are found to be duplicates, expect for the removed transaction)

**Extra info**
I understand it is tricky to automate, because we are talking about a system were the total is only known after 2 imports.
Maybe I should make a rule that creates a transfer for all "expense" transactions that are for known accounts, and that ignores all "income" transfers for the same? That way I always ignore the 'extra' transaction, and automatically change transactions into transfers?

**Comment by JC5** [comment]
Interesting problem, but it's not one without a solution.

First of all, when you import from either bank, make sure you link the opposing account to the correct asset account. Use the "map this value" to do this, and make sure you do this for both then opposing account name and the opposing account IBAN.

If you do this, and Firefly III encounters a transaction where both the source (your ING account) and the destination (the ABN account) are asset asccounts, it will automatically create transfers.

This should already remove one problem. Importing from either the first or second account should result in transfers. To make sure this works, give your asset accounts the correct IBAN values.

The second part is more tricky. Firefly III merge transfers (aka ignore the second one), but only if the dates are the same, the description is exactly the same, the amount is the same and the asset accounts are the same (but in reverse).

So, if you can import transfers and IBN and ABN agree on the description it should work fine.

**Comment by NanneHuiges** [comment]
Right, so the first step should be the mapping! That would be the 'trigger' that makes it a transfer? Great!

Can you map these values after an import, or should I make sure to change them during import? Anything I can do after-import?

**Comment by JC5** [comment]
I'm afraid not, there is no mass convert option. If you want to re-import keep in mind that you must delete the entries from the database or they will be skipped.

**Comment by NanneHuiges** [comment]
"no mass convert" implies that I can do it for a single one? Because I couldn't find the option to do that. (sorry to be a bother btw, just wrapping my head around this :D )

I'll do it next time round to test out the transfers for sure.

**Comment by JC5** [comment]
Yes, just go to `/transactions/show/xxx` and check the menu between edit and delete.

**Comment by NanneHuiges** [comment]
Ow wait, you mean just transform them into a transfer because that's basically the same as mapping the data, right? Looks good, thanks!
